### PR TITLE
AP_GPS: Support SBF GPS Yaw message

### PR DESCRIPTION
**Pull request by Vkd4U2NtVkZOVg**

### CODE REVIEW

**Comment by Vkd4U2NtVkZOVg** [comment]
@tridge Could you please review this?

**Comment by tatsuy** [comment]
We already have a PR (#12429) that includes the same functionality.

File: libraries/AP_GPS/AP_GPS_SBF.cpp [hunk 1]
```diff
@@ -496,6 +495,30 @@ AP_GPS_SBF::process_message(void)
...
+            state.gps_yaw = (float)(temp.Heading);
+            state.have_gps_yaw_accuracy = true;
+            break;
+        } else {
+            state.have_gps_yaw = false;
+            state.gps_yaw_configured = false;
+            state.have_gps_yaw_accuracy = false;
+            break;
+        }
+    }
```
Comment (Abner0907) [inline libraries/AP_GPS/AP_GPS_SBF.cpp:521]: why not ?
```
if (temp.Mode && (temp.Heading > -200000)) {
            state.have_gps_yaw = true;
            state.gps_yaw_configured = true;
            state.gps_yaw = (float)(temp.Heading);
            state.have_gps_yaw_accuracy = true;
            break;
      } else {
          state.have_gps_yaw = false;
          state.gps_yaw_configured = false;
          state.have_gps_yaw_accuracy = false;
          break;
      }
```

File: libraries/AP_GPS/AP_GPS_SBF.h [hunk 2]
```diff
@@ -107,7 +107,8 @@ class AP_GPS_SBF : public AP_GPS_Backend
         PVTGeodetic = 4007,
         ReceiverStatus = 4014,
         BaseVectorGeod = 4028,
-        VelCovGeodetic = 5908
+        VelCovGeodetic = 5908,
+        AttEuler       = 5938
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS_SBF.h:111]:
```suggestion
        AttEuler       = 5938,
```

Adding the comma just keeps future diffs shorter.

**Review by WickedShell — CHANGES_REQUESTED** [review]
@Vkd4U2NtVkZOVg How are you handling the configuration string? AttEuler should be removed from the messages being sent at the moment if auto configuration is enabled (which it should be). I think we will need a flag to the GPS constructor to indicate we want to use GPS attitude and add AttEuler to the config string. That should be a pretty small addition, but I'm not sure how this was tested without that.

**Comment by WickedShell** [comment]
> We already have a PR (#12429) that includes the same functionality.

That PR is a bit stalled. The most up to date is actually my branch that's in a comment, but the SBF driver underwent a signficant rewrite since then to the point that adding from scratch may be the easier path at the moment.

**Comment by Vkd4U2NtVkZOVg** [comment]
@WickedShell I just disable the autoconfig for now, the only reason to add AttEuler is we want to use the gps yaw, so can we use have_gps_yaw flag and just add the AttEuler to config string?

**Comment by WickedShell** [comment]
Yes, you will want the covariance to get the yaw accuracy, as it makes a difference within the data processing.


>I just disable the autoconfig for now, the only reason to add AttEuler is we want to use the gps yaw, so can we use have_gps_yaw flag and just add the AttEuler to config string?

I'm pretty sure we can't always add the AttEuler, because I don't think it's always enabled or known. You will need a new GPS type (SBF_With_Yaw) and then use that to flag that the AttEuler should be added to the config string.

**Comment by Vkd4U2NtVkZOVg** [comment]
@WickedShell Thanks, I think It's the good way. And also I try to add the novatel's dual antenna GPS support GPS yaw, I could use the same way to do that!

**Pull request closed without merging**